package.json

```
@@ -13,16 +11,20 @@
     "node": ">=16.18.0"
   },
   "scripts": {
-    "start": "tsdx watch",
-    "build": "tsdx build",
-    "test": "tsdx test",
-    "test:watch": "tsdx test --watch",
-    "test:e2e": "tsdx test --config jest-e2e-config.json",
+    "start": "nodemon -e ts,tsx,json,js,jsx --watch ./src --ignore ./dist --exec 'yarn build --incremental'",
+    "build": "yarn run clean && yarn run build:cjs && yarn run build:esm && yarn run build:types",
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --outDir ./dist/cjs --removeComments --verbatimModuleSyntax false && echo > ./dist/cjs/package.json '{\"type\":\"commonjs\"}'",
+    "build:esm": "tsc --project tsconfig.build.json --module es2015 --outDir ./dist/esm && echo > ./dist/esm/package.json '{\"type\":\"module\",\"sideEffects\":false}'",
+    "build:types": "tsc --project tsconfig.build.json --module esnext --declarationDir ./dist/types --emitDeclarationOnly --declaration --declarationMap",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:e2e": "jest --config jest-e2e-config.json",
     "lint": "eslint --fix src",
     "lint-check": "eslint src",
-    "prepare": "tsdx build && husky install",
+    "prepare": "tsc --project ./tsconfig.build.json && husky install",
     "size": "size-limit",
     "analyze": "size-limit --why",
+    "clean": "rimraf dist",
```

Are there advantages to using rimraf? I don't see it as a direct dependency, so we're inheriting it from some other dependency.

FWIW, chaining `mv ... && rm -rf ... &` is usually pretty quick because it cleans up with having to wait on the deletion to finish.

```suggestion
    "clean": "mv dist .dist && rm -rf .dist &",
```

---

The primary advantage is that the module is agnostic of OS. The command effectively has fallbacks for POSIX & Windows

---

We don't target Windows as a platform and rm is POSIX, so is it worth depending on a package just for this?

---

Done https://github.com/across-protocol/sdk-v2/pull/382/commits/80599f4aa5d76861e754bc034267dfdaf79edeff

I've kept this as a synchronous call since we'll need to ensure dist is deleted before rebuilding